doc/internal/man7/EVP_PKEY.pod

```diff
@@ -8,37 +8,198 @@ EVP_PKEY - an internal description
+
+This is especially visible with internal legacy keys, since they can
+be extracted with functions like L<EVP_PKEY_get0_RSA(3)> and then
+modified at will with functions like L<RSA_set0_key(3)>.
+
+Internal provider native keys are also possible to be modified, if the
+associated L<EVP_KEYMGMT(3)> implementation allows it.  This is done
+with L<EVP_PKEY_set_params(3)> and its specialised derivatives.  The
+OpenSSL providers allow it for the following:
+
+=over 4
+
+=item DH, EC, ED25519, ED448, X25519, X448:
+
+It's possible to set the encoded public key.  This is supported in
+particular through L<EVP_PKEY_set1_encoded_public_key(3)>.
+
+=item EC:
+
+It's possible to flip the ECDH cofactor mode.
+
+=back
+
+Every time the B<EVP_PKEY> internal key mutates, an internal dirty
+count is incremented.  The need for a dirty count is explained further
+in L</Export cache for provider operations>.
+
+For provider native origin keys, this doesn't require any help from
+the L<EVP_KEYMGMT(3)>, the dirty count is maintained in the B<EVP_PKEY>
+itself, and is incremented every time L<EVP_PKEY_set_params(3)> or its
+specialised derivatives are called.
+For legacy origin keys, this requires the associated
+L<EVP_PKEY_ASN1_METHOD(3)> to implement the dirty_cnt() function.  All
+of OpenSSL's built-in L<EVP_PKEY_ASN1_METHOD(3)> implement this
+function.
+
+=head2 Export cache for provider operations
+
+OpenSSL 3.0 can handle operations such as signing, encrypting, etc in
+diverse providers, potentially others than the provider of the
+L<EVP_KEYMGMT(3)>.  Since two providers, possibly from different
+vendors, can't be expected to share internal key structures.  There
+are therefore instances where key data will need to be exported to the
+provider that is going to perform the operation (this also implies
+that every provider that implements a key pair based operation must
+also implement am L<EVP_KEYMGMT(3)>).
+
+For performance reasons, libcrypto tries to minimize the need to
+perform such an export, so it maintains a cache of such exports in the
+B<EVP_PKEY>.  Each cache entry has two items, a pointer to the
+provider side key data and the associated L<EVP_KEYMGMT(3)>.
+
+I<This cache is often referred to as the "operation key cache", and
+the key data that the cached keys came from as the "origin", and since
+there are two forms of the latter, we have the "legacy origin" and the
+"provider native origin".>
+
+The export to the operation key cache can be performed independent of
+what form the origin has.
+For a legacy origin, this requires that the associated
+L<EVP_PKEY_ASN1_METHOD(3)> implements the functions export_to() and
+dirty_cnt().
+For a provider native origin, this requires that the associated
+L<EVP_KEYMGMT(3)> implements the OSSL_FUNC_keymgmt_export() function
+(see L<provider-keymgmt(7)>).
+In all cases, the receiving L<EVP_KEYMGMT(3)> (the one associated with
+the exported key data) must implement OSSL_FUNC_keygmt_import().
```

Comment:
Missing an m after key?

Reply:
Why doesn't doc-nits complain about such things?

Reply:
Yes.  Fixed

Reply:
> Why doesn't doc-nits complain about such things?

How can it tell that OSSL_FUNC_keygmt_import() is a typo, and not a reference to some other function in some other library? Implementing that kind of spell-check would be costly and if done should be a separate program.

Reply:
It could extract the functions from doc/man7/provider-NAME.pod...

The reason we don't check is that section 7 manuals aren't programming manuals, they are "misc".  The reason why we don't have this file, or doc/man7/provider-NAME.pod, in section 3 is that they aren't public interfaces per se.

Reply:
> It could extract the functions from doc/man7/provider-NAME.pod...

That's not a general solution.  You are then embedding too much semantic knowledge into what is a syntax-checker.

Reply:
You're quite right about that...  so I think that for the moment, I'll go with "too hard to do, post-pone for another rainy day"